lib/marshalingCommands.js

```diff
@@ -0,0 +1,173 @@
+var _ = require('lodash');
+var unexpectedItemHandler = require('./unexpectedItemHandler');
+
+/**
+ * Converts boolean value to DynamoDb "BOOL"
+ *
+ * @param item
+ * @returns {*}
+ */
+function marshalBoolean(item) {
+  if (!_.isBoolean(item)) {
+    return undefined;
+  }
+
+  return {BOOL: item};
+}
+
+/**
+ * Converts mixed array to DynamoDb "L"
+ *
+ * @param item
+ * @param marshal
+ * @returns {*}
+ */
+function marshalList(item, marshal) {
+  if (!_.isArray(item)) {
+    return undefined;
+  }
+
+  return {L: _.map(item, marshal)};
+}
+
+/**
+ * Converts object literal to DynamoDb "M"
+ *
+ * @param item
+ * @param marshal
+ * @returns {*}
+ */
+function marshalMap(item, marshal) {
+  // Todo make this accept ES6 Map or Object Literal
+  if (!_.isPlainObject(item)) {
+    return undefined;
+  }
+
+  return {
+    M: _.mapValues(item, function(value) {
+      return marshal(value);
+    })
+  };
+}
+
+/**
+ * Converts null to DynamoDb "NULL"
+ *
+ * @param item
+ * @returns {*}
+ */
+function marshalNull(item) {
+  if (!_.isNull(item) && !_.isUndefined(item)) {
+    return undefined;
+  }
+
+  return {NULL: true};
+}
+
+/**
+ * Converts number to DynamoDb "N"
+ *
+ * @param item
+ * @returns {*}
+ */
+function marshalNumber(item) {
+  if (!_.isNumber(item)) {
+    return undefined;
+  }
+
+  return {N: item.toString()};
+}
+
+/**
+ * Helper function for arrays to DynamoDb "NS"
+ *
+ * @param arr
+ * @returns {*}
+ */
+function marshalArrayToNumberSet(arr) {
+  if (
+    _.isEmpty(arr) ||
```

Comment:
The isEmpty check? Or the whole if condition?

The difference in conditions is just that one checks for arrays of unique strings, and the other checks for arrays of unique numbers.